src/providers/db2/qgsdb2newconnection.cpp

```diff
@@ -147,7 +145,7 @@ void QgsDb2NewConnection::on_cb_trustedConnection_clicked()
 
 QgsDb2NewConnection::~QgsDb2NewConnection()
 {
-
+  delete bar;
```

Comment:
Where is this created? In the .ui file?
Normally there's no need for manual cleanup with widgets if they are properly parented.